src/rubrix/labeling/text_classification/weak_labels.py

```diff
@@ -517,18 +617,343 @@ def change_mapping(self, label2int: Dict[str, int]):
                 )
             # compute masks
             label_masks[label] = self._matrix == self._label2int[label]
-            annotation_masks[label] = self._annotation_array == self._label2int[label]
+            annotation_masks[label] = self._annotation == self._label2int[label]
 
         # swap integers
         for label in self._label2int:
             self._matrix[label_masks[label]] = label2int[label]
-            self._annotation_array[annotation_masks[label]] = label2int[label]
+            self._annotation[annotation_masks[label]] = label2int[label]
 
         # update mapping dicts
         self._label2int = label2int.copy()
         self._int2label = {val: key for key, val in self._label2int.items()}
 
 
+class WeakMultiLabels(WeakLabelsBase):
+    """Computes the weak labels of a multi-label text classification dataset by applying a given list of rules.
+
+    Args:
+        dataset: Name of the dataset to which the rules will be applied.
+        rules: A list of rules (labeling functions). They must return a string, or ``None`` in case of abstention.
+            If None, we will use the rules of the dataset (Default).
+        ids: An optional list of record ids to filter the dataset before applying the rules.
+        query: An optional ElasticSearch query with the
+            `query string syntax <https://rubrix.readthedocs.io/en/stable/reference/webapp/search_records.html>`_
+            to filter the dataset before applying the rules.
+
+    Raises:
+        NoRulesFoundError: When you do not provide rules, and the dataset has no rules either.
+        DuplicatedRuleNameError: When you provided multiple rules with the same name.
+        SingleLabelError: When trying to get weak labels for a single-label text classification task.
+        NoRecordsFoundError: When the filtered dataset is empty.
+
+    Examples:
+        >>> # Get the 3 dimensional weak label matrix from a multi-label classification dataset with rules:
+        >>> weak_labels = WeakMultiLabels(dataset="my_dataset")
+        >>> weak_labels.matrix()
+        >>> weak_labels.summary()
+        >>>
+        >>> # Get the 3 dimensional weak label matrix from rules defined in Python:
+        >>> def awesome_rule(record: TextClassificationRecord) -> str:
+        ...     return ["Positive", "Slang"] if "next level" in record.inputs["text"] else None
+        >>> another_rule = Rule(query="amped OR psyched", label=["Positive", "Slang"])
+        >>> weak_labels = WeakMultiLabels(dataset="my_dataset", rules=[awesome_rule, another_rule])
+        >>> weak_labels.matrix()
+        >>> weak_labels.summary()
+    """
+
+    def __init__(
+        self,
+        dataset: str,
+        rules: Optional[List[Callable]] = None,
+        ids: Optional[List[Union[int, str]]] = None,
+        query: Optional[str] = None,
+    ):
+        super().__init__(dataset=dataset, rules=rules, ids=ids, query=query)
+
+        if not self._records[0].multi_label:
```

Comment:
This check means that the WeakMultiLabels can only be generated from multi-labels datasets.

Do you think it would be interesting to allow applying it to single-label datasets? We have some use case where that would be useful.

Reply:
Removed check